includes/admin/class-wp-job-manager-settings.php

```diff
@@ -303,6 +303,15 @@ protected function init_settings() {
 								'url'   => __( 'Website URLs only', 'wp-job-manager' ),
 							],
 						],
+						[
+							'name'       => 'job_manager_show_agreement_job_submission',
+							'std'        => '0',
+							'label'      => __( 'Agreement checkbox', 'wp-job-manager' ),
+							'cb_label'   => __( 'Enable a required agreement checkbox on the form', 'wp-job-manager' ),
+							'desc'       => __( 'Show a checkbox that needs to be marked in order to submit the form. You can also add a link to Terms and Conditions by selecting the respective page in the "Pages" tab. And the label content can be customized using the <code>job_manager_agreement_label</code> filter.', 'wp-job-manager' ),
```

Comment:
I'm not sure we need the bit about the filter. What about this for a description?
```
Require a Terms and Conditions checkbox to be marked before a job can be submitted. The linked page can be set from the <a href="#settings-job_pages">Pages</a> settings tab.
```

Reply:
https://github.com/Automattic/WP-Job-Manager/pull/2155/commits/ea513f367b0e59288a60c7272c5b7ea9d450f545